kayenta-newrelic-insights/src/main/java/com/netflix/kayenta/newrelic/service/NewrelicTimeSeries.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.kayenta.newrelic.service;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Stream;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Data
+public class NewrelicTimeSeries {
+
+  private List<NewrelicSeriesEntry> timeSeries;
+  private NewresultResultMetadata metadata;
+
+  @Data
+  static public class NewrelicSeriesEntry {
+
+    private Long beginTimeSeconds;
+    private Long inspectedCount;
+    private Long endTimeSeconds;
+    private List<HashMap<String, Number>> results;
+
+    @JsonIgnore
+    private Double adjustSingleResult(HashMap<String, Number> entry) {
```

Comment:
Does New Relic return sparse results, or is it guaranteed to return a complete set of data with missing data set to some sort of empty value?

Exmple of sparse result for some gauge:

```json
{
	"1234": 1,
	"1236": 1,
	"1237": 1
}
```

Notice that that timestamp ```1235``` is missing

example of complete result:

```json
{
	"1234": 1,
	"1235": null,
	"1236": 1,
	"1237": 1
}
```

In either example if the step (data resolution) was set to `1` the following data would be expected.

```json
[1,NaN,1,1]
```

Reply:
- [ ] Sanity check that New Relic doesn't return sparse data.

Reply:
New Relic returns complete results. null-values are converted to NaN by our extension.